src/BitNode/ui/BitNodeAdvancedOptions.tsx

```diff
@@ -0,0 +1,355 @@
+  you enter BN 1.3 while having SF 1.2 but with the active level set to 0, you will not get the bonuses from SF
+  1.1 or SF 1.2, but you will still earn SF 1.3 when destroying the BN.`;
+  const note = currentSourceFiles.has(10) ? (
+    <>
+      <Typography>Note:</Typography>
+      <ul style={{ marginTop: 0 }}>
+        <li>{basicNote}</li>
+        <li>
+          Changing the active level of SF 10 does not affect your current sleeves or the maximum number of sleeves.
+        </li>
+      </ul>
+    </>
+  ) : (
+    <>
+      <Typography>Note: {basicNote}</Typography>
+      <br />
+    </>
+  );
+
+  return (
+    <>
+      <Typography>Override active level of Source-File:</Typography>
+      <br />
+      <Typography component="div">{note}</Typography>
+      <div>
+        <Select
+          disabled={menuItemList.length === 0}
+          value={selectElementValue ?? ""}
+          onChange={(event) => {
+            setSelectElementValue(Number(event.target.value));
+          }}
+          sx={{ minWidth: "80px" }}
+        >
+          {menuItemList.map((sfNumber) => (
+            <MenuItem key={sfNumber} value={sfNumber}>
+              SF-{sfNumber}
+            </MenuItem>
+          ))}
+        </Select>
+        <Button
+          disabled={menuItemList.length === 0}
+          onClick={() => {
+            if (!selectElementValue) {
+              return;
+            }
+            setSfOverrides((old) => {
+              const newSfOverrides = new JSONMap(old);
+              newSfOverrides.set(selectElementValue, getSfLevel(selectElementValue));
+              const newMenuItemList = getMenuItemList(newSfOverrides);
+              if (newMenuItemList.length > 0) {
+                setSelectElementValue(newMenuItemList[0]);
+              } else {
+                setSelectElementValue(null);
+              }
+              return newSfOverrides;
+            });
+          }}
+          sx={{ marginLeft: "1rem" }}
+        >
+          Add
+        </Button>
+        <ButtonWithTooltip
+          normalTooltip="Remove all overridden SF"
+          disabledTooltip={sfOverrides.size === 0 ? "No overridden SF" : ""}
+          onClick={() => {
+            setSfOverrides(new JSONMap());
+            setSelectElementValue(firstSourceFile);
+            callbacks.resetSourceFileOverrides();
+          }}
+          buttonProps={{ sx: { marginLeft: "1rem" } }}
+        >
+          Remove all
+        </ButtonWithTooltip>
+      </div>
+      <br />
+      {sfOverrides.size > 0 && (
+        <>
+          <table>
+            <tbody>
+              <tr>
+                <td>
+                  <Tooltip title="Set active level for all chosen SF">
+                    <Typography minWidth="7rem">Set all SF</Typography>
+                  </Tooltip>
+                </td>
+                <td>
+                  <ButtonGroup>
+                    {Array.from([0, 1, 2, 3]).map((level) => (
```

Comment:
`[0, 1, 2, 3]` is already an array
```suggestion
                    {[0, 1, 2, 3].map((level) => (
```